### DISCUSSION TITLE
Task Lists: API

### DISCUSSION THREAD

**Discussion by abernix** [community/community#4935] [Projects and Issues]
It would be great if the GitHub Issues GraphQL APIs (or REST!) surfaced task lists in a structured manner, particularly now that they can be broken into subtasks.  This seems like it'd be a good opportunity to use the [`IssueConnection`](https://docs.github.com/en/graphql/reference/objects#issueconnection) object type, much in the same way that it's used for the [`closingIssuesReferences` field on the `PullRequest` object](https://docs.github.com/en/graphql/reference/objects#pullrequest).

e.g., in GraphQL this might be:

> Note: Just an example!

```graphql
query {
  repository(owner:"org-name" name) {
    issue(number: 14){
      taskListIssuesReferences(first:10) {
        nodes {
          id
          title # This would be a task list issue.
        }
      }
    }
  }
}
```

Right now, I think the option is to parse the `body` or `bodyText` properties for `- [ ] org/repo#123` references?

Admittedly, perhaps `IssueConnection` is just one of a couple objects that might be better returned from a `TaskList` object that might have task list items that are not yet converted to issues.  For example, in the case of `- [ ] org/repo#123 org/repo#456 additional text`, I could imagine a `TaskList` that just returns an `IssueConnection` containing both `org/repo#123` _and_ `org/repo#456`, but also just the raw text too (`org/repo#123 org/repo#456 additional text`). 

---

In the same spirit, if an Issue is part of a task-list, the GitHub UI shows that it's _tracked_ in a task list of a parent issue, and it would be nice to surface that detail in the API, too.

> Note: Again, just an example right now!

```graphql
query {
  repository(owner:"org-name" name) {
    issue(number: 14){
      trackedInIssuesReferences(first:10) {
        nodes {
          id
          # This would be a parent issue which has this issue in its task list
          title
        }
      }
    }
  }
}
```

I'm not sure if there's any easy workaround to find "tracked in" from the API right now?

**Comment by blqke** [comment]
++

I am on this exact subject too. About "knowing the child issues" I didn't find an easy workaround yet so I will stick with parsing the `body` field, but it is not very convenient.

About "knowing the parent issue", as you said Github UI is cleary showing this ⏬ so this feature clearly exists somewhere outside the GraphQL API.
![image](https://user-images.githubusercontent.com/1248148/128631770-f7e2d480-4945-4865-80af-4a03d7707097.png)
